browser/src/map/Clipboard.js

```diff
@@ -876,55 +879,60 @@ L.Clipboard = L.Class.extend({
 		// that command so we are sure the clipboard is set before
 		// fetching it.
 
-		const that = this;
-
-		if (that._commandCompletion.length > 0)
-			window.app.console.error('Already have ' + that._commandCompletion.length +
+		if (this._commandCompletion.length > 0)
+			window.app.console.error('Already have ' + this._commandCompletion.length +
 						 ' pending clipboard command(s)');
 
-		const url = that.getMetaURL() + '&MimeType=text/html,text/plain;charset=utf-8';
-
-		// Share a single fetch
-		var fetchPromise = function() {
-			return new Promise((resolve, reject) => {
-				try {
-					var result = fetch(url).then(response => response.text());
-					resolve(result);
-				} catch (err) {
-					reject(err);
-				}
+		const url = this.getMetaURL() + '&MimeType=text/html,text/plain;charset=utf-8';
+		const commandCompletePromise = new Promise((resolve, reject) => {
+			window.app.console.log('New ' + command + ' promise');
+			// FIXME: add a timeout cleanup too ...
+			this._commandCompletion.push({
+				resolve: resolve,
+				reject: reject,
 			});
-		};
+		});
+
+		await commandCompletePromise;
 
-		var awaitPromise = function(url, mimetype, shorttype) {
-			return new Promise((resolve, reject) => {
-				window.app.console.log('New ' + command + ' promise');
-				// FIXME: add a timeout cleanup too ...
-				that._commandCompletion.push({ fetch: fetchPromise, command: command,
-							       resolve: resolve, reject: reject,
-							       mimetype: mimetype, shorttype: shorttype});
-		}); };
-
-		const text = new ClipboardItem({
-			'text/html': awaitPromise(url, 'text/html', 'html'),
-			'text/plain': awaitPromise(url, 'text/plain', 'plain')
+		const response = await fetch(url);
+		const responseText = await response.text();
+
+		const clipboardItem = new ClipboardItem({
+			'text/plain': this._parseClipboardFetchResult(responseText, 'text/plain', 'plain'),
+			'text/html': this._parseClipboardFetchResult(responseText, 'text/html', 'html'),
 		});
 		let clipboard = navigator.clipboard;
 		if (L.Browser.cypressTest) {
 			clipboard = this._dummyClipboard;
 		}
-		clipboard.write([text]).then(function() {
-		}, function(error) {
+
+		try {
+			await clipboard.write([clipboardItem]);
+		} catch (error) {
 			window.app.console.log('navigator.clipboard.write() failed: ' + error.message);
 
 			// Warn that the copy failed.
-			that._warnCopyPaste();
+			this._warnCopyPaste();
 			// Once broken, always broken.
 			L.Browser.clipboardApiAvailable = false;
-			window.prefs.set('clipboardApiAvailable', 'false');
+			window.prefs.set('clipboardApiAvailable', false);
```

Comment:
previously it was string not boolean (`false`)

Reply:
Correct. I wrote the prefs API and these do the same thing (we convert everything that goes in to a string), but since as we are getting booleans later I feel it is clearer to do this ... it's also more consistent with other uses of the prefs API for storing booleans (e.g. see some of the stuff on the dark mode). As I was refactoring this code anyway, I decided to change the usage at the same time

![image](https://github.com/user-attachments/assets/5f40f93e-74fe-4a65-9d1c-3f5b6c36723c)

Reply:
ok fine, I just note things I found different so you can check if they are intentional :)

Reply:
all good, I see it as a non-functional change but I agree it does look like it could behave differently :)